### PR TITLE
Fix reward distribution itests

### PR DESCRIPTION
**Pull request by irina-pereiaslavskaia**
Fix rounding in reward test data.

### CODE REVIEW

File: itests/testdata/reward_distribution_rolllback_testdata.go [hunk 1]
```diff
@@ -23,13 +25,13 @@
...
 ) RewardDistributionTestData[RewardDistributionExpectedValues] {
-	currentReward := int64(utl.GetCurrentReward(suite, height))
+	currentReward := utl.GetCurrentReward(suite, height)
 	return NewRewardDistributionTestData(
 		addresses,
 		RewardDistributionExpectedValues{
-			MinersSumDiffBalance: currentReward - 2*currentReward/3,
-			DaoDiffBalance:       currentReward / 3,
-			XtnDiffBalance:       currentReward / 3,
+			MinersSumDiffBalance: int64(currentReward) - int64(2*math.Round(float64(currentReward)/3)),
```
[the remark this answers is not in the document]
Comment (alexeykiselev) [reply itests/testdata/reward_distribution_rolllback_testdata.go:32]: I think we don't need `math` here, it's just an incorrect order of calculations. See example https://go.dev/play/p/pM58zeXjXbG

File: itests/testdata/reward_distribution_rolllback_testdata.go [hunk 2]
```diff
@@ -40,13 +42,13 @@ func GetRollbackBeforeF19TestData(
...
+	currentReward := utl.GetCurrentReward(suite, height)
 	return NewRewardDistributionTestData(
 		addresses,
 		RewardDistributionExpectedValues{
-			MinersSumDiffBalance: currentReward / 3,
-			DaoDiffBalance:       currentReward / 3,
-			XtnDiffBalance:       currentReward / 3,
+			MinersSumDiffBalance: int64(math.Round(float64(currentReward) / 3)),
+			DaoDiffBalance:       int64(math.Round(float64(currentReward) / 3)),
+			XtnDiffBalance:       int64(math.Round(float64(currentReward) / 3)),
```
Comment (alexeykiselev) [inline itests/testdata/reward_distribution_rolllback_testdata.go:51]: In this case we have to calculate as follows:
```go
p := currentReward / 3
...
                        MinersSumDiffBalance: currentReward - 2*p,
			DaoDiffBalance:       p,
			XtnDiffBalance:       p,
```
actually, the same way as above.

**Review by alexeykiselev — CHANGES_REQUESTED** [review]

File: itests/testdata/reward_distribution_rolllback_testdata.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (irina-pereiaslavskaia) [reply itests/testdata/reward_distribution_rolllback_testdata.go:32]: Fix.

File: itests/testdata/reward_distribution_rolllback_testdata.go (hunk 2 above)
Comment (irina-pereiaslavskaia) [reply itests/testdata/reward_distribution_rolllback_testdata.go:51]: Fix

**Review by alexeykiselev — APPROVED** [review]

**Review by nickeskov — APPROVED** [review]

**Review by nickeskov — APPROVED** [review]

**Pull request merged**